src/workerd/api/sockets.h

```diff
@@ -18,16 +18,30 @@ struct SocketAddress {
 
 struct SocketOptions {
   jsg::Unimplemented tls; // TODO(later): TCP socket options need to be implemented.
-  JSG_STRUCT(tls);
+  bool allowHalfOpen;
```

Comment:
Best to make this either `jsg::Optional` or explicitly default to `false`, just for clarity...

```
jsg::Optional<bool> allowHalfOpen;

// or

bool allowHalfOpen = false;
```